File: samples/client/petstore/java/src/main/java/io/swagger/client/api/PetApi.java
package io.swagger.client.api;

import io.swagger.client.ApiException;
import io.swagger.client.ApiClient;
import io.swagger.client.Configuration;

import io.swagger.client.model.*;

import java.util.*;

import io.swagger.client.model.Pet;
import java.io.File;

import org.glassfish.jersey.media.multipart.FormDataBodyPart;
import org.glassfish.jersey.media.multipart.FormDataMultiPart;
import org.glassfish.jersey.media.multipart.MultiPart;
import org.glassfish.jersey.media.multipart.file.FileDataBodyPart;

import javax.ws.rs.client.Entity;
import javax.ws.rs.core.Form;
import javax.ws.rs.core.MediaType;

import java.io.File;
import java.util.Map;
import java.util.HashMap;

public class PetApi {
  private ApiClient apiClient;

  public PetApi() {
    this(Configuration.getDefaultApiClient());
  }

  public PetApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Update an existing pet
   * 
   * @param body Pet object that needs to be added to the store
   * @return void
   */
  public void updatePet (Pet body) throws ApiException {
    Object postBody = body;
    

    // create path and map variables
    String path = "/pet".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json", "application/xml"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "PUT", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Add a new pet to the store
   * 
   * @param body Pet object that needs to be added to the store
   * @return void
   */
  public void addPet (Pet body) throws ApiException {
    Object postBody = body;
    

    // create path and map variables
    String path = "/pet".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json", "application/xml"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Finds Pets by status
   * Multiple status values can be provided with comma seperated strings
   * @param status Status values that need to be considered for filter
   * @return List<Pet>
   */
  public List<Pet> findPetsByStatus (List<String> status) throws ApiException {
    Object postBody = null;
    

    // create path and map variables
    String path = "/pet/findByStatus".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    if (status != null)
      queryParams.put("status", apiClient.parameterToString(status));
    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return (List<Pet>) apiClient.deserialize(response, "array", Pet.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Finds Pets by tags
   * Muliple tags can be provided with comma seperated strings. Use tag1, tag2, tag3 for testing.
   * @param tags Tags to filter by
   * @return List<Pet>
   */
  public List<Pet> findPetsByTags (List<String> tags) throws ApiException {
    Object postBody = null;
    

    // create path and map variables
    String path = "/pet/findByTags".replaceAll("\\{format\\}","json");

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    if (tags != null)
      queryParams.put("tags", apiClient.parameterToString(tags));
    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return (List<Pet>) apiClient.deserialize(response, "array", Pet.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Find pet by ID
   * Returns a pet when ID &lt; 10.  ID &gt; 10 or nonintegers will simulate API error conditions
   * @param petId ID of pet that needs to be fetched
   * @return Pet
   */
  public Pet getPetById (Long petId) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'petId' is set
    if (petId == null) {
       throw new ApiException(400, "Missing the required parameter 'petId' when calling getPetById");
    }
    

    // create path and map variables
    String path = "/pet/{petId}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "petId" + "\\}", apiClient.escapeString(petId.toString()));

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "api_key", "petstore_auth" };
      String response = apiClient.invokeAPI(path, "GET", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return (Pet) apiClient.deserialize(response, "", Pet.class);
      }
      else {
        return null;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Updates a pet in the store with form data
   * 
   * @param petId ID of pet that needs to be updated
   * @param name Updated name of the pet
   * @param status Updated status of the pet
   * @return void
   */
  public void updatePetWithForm (String petId, String name, String status) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'petId' is set
    if (petId == null) {
       throw new ApiException(400, "Missing the required parameter 'petId' when calling updatePetWithForm");
    }
    

    // create path and map variables
    String path = "/pet/{petId}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "petId" + "\\}", apiClient.escapeString(petId.toString()));

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/x-www-form-urlencoded"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if (name != null) {
        hasParams = true;

        FormDataMultiPart mp = new FormDataMultiPart();
        mp.bodyPart(new FormDataBodyPart("name", apiClient.parameterToString(name)));
        multipart.bodyPart(mp, MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (status != null) {
        hasParams = true;

        FormDataMultiPart mp = new FormDataMultiPart();
        mp.bodyPart(new FormDataBodyPart("status", apiClient.parameterToString(status)));
        multipart.bodyPart(mp, MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      if (name != null) {
        form.param("name", apiClient.parameterToString(name));
        hasValues = true;
      }
      if (status != null) {
        form.param("status", apiClient.parameterToString(status));
        hasValues = true;
      }
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * Deletes a pet
   * 
   * @param apiKey 
   * @param petId Pet id to delete
   * @return void
   */
  public void deletePet (String apiKey, Long petId) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'petId' is set
    if (petId == null) {
       throw new ApiException(400, "Missing the required parameter 'petId' when calling deletePet");
    }
    

    // create path and map variables
    String path = "/pet/{petId}".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "petId" + "\\}", apiClient.escapeString(petId.toString()));

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    if (apiKey != null)
      headerParams.put("api_key", apiClient.parameterToString(apiKey));
    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "DELETE", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
  /**
   * uploads an image
   * 
   * @param petId ID of pet to update
   * @param additionalMetadata Additional data to pass to server
   * @param file file to upload
   * @return void
   */
  public void uploadFile (Long petId, String additionalMetadata, File file) throws ApiException {
    Object postBody = null;
    
    // verify the required parameter 'petId' is set
    if (petId == null) {
       throw new ApiException(400, "Missing the required parameter 'petId' when calling uploadFile");
    }
    

    // create path and map variables
    String path = "/pet/{petId}/uploadImage".replaceAll("\\{format\\}","json")
      .replaceAll("\\{" + "petId" + "\\}", apiClient.escapeString(petId.toString()));

    // query params
    Map<String, String> queryParams = new HashMap<String, String>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Entity<?> formEntity = null;

    

    

    final String[] accepts = {
      "application/json", "application/xml"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "multipart/form-data"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    if(contentType.startsWith("multipart/form-data")) {
      boolean hasParams = false;
      MultiPart multipart = new MultiPart();

      
      if (additionalMetadata != null) {
        hasParams = true;

        FormDataMultiPart mp = new FormDataMultiPart();
        mp.bodyPart(new FormDataBodyPart("additionalMetadata", apiClient.parameterToString(additionalMetadata)));
        multipart.bodyPart(mp, MediaType.MULTIPART_FORM_DATA_TYPE);
      }
      
      if (file != null) {
        hasParams = true;
        multipart.bodyPart(new FileDataBodyPart("file", new java.io.File(file.getName()), MediaType.APPLICATION_OCTET_STREAM_TYPE));
      }
      
      if(hasParams)
        formEntity = Entity.entity(multipart, MediaType.MULTIPART_FORM_DATA_TYPE);
    }
    else {
      boolean hasValues = false;
      Form form = new Form();
      if (additionalMetadata != null) {
        form.param("additionalMetadata", apiClient.parameterToString(additionalMetadata));
        hasValues = true;
      }
      
      
      if(hasValues) {
        formEntity = Entity.entity(form, MediaType.APPLICATION_FORM_URLENCODED_TYPE);
      }
    }

    try {
      String[] authNames = new String[] { "petstore_auth" };
      String response = apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formEntity, accept, contentType, authNames);
      if(response != null){
        return ;
      }
      else {
        return ;
      }
    } catch (ApiException ex) {
      throw ex;
    }
  }
  
}

File: samples/clientpetstorecsharp/src/main/csharp/io/swagger/Api/PetApi.cs
using System;
using System.Collections.Generic;
using io.swagger.client;
using io.swagger.Model;







namespace io.swagger.Api {
  
  public class PetApi {
    private readonly ApiInvoker _apiInvoker = ApiInvoker.GetInstance();

    public PetApi(String basePath = "http://petstore.swagger.io/v2")
    {
      BasePath = basePath;
    }

    public ApiInvoker GetInvoker() {
      return _apiInvoker;
    }

    public string BasePath { get; set; }

    

    /// <summary>
    /// Update an existing pet 
    /// </summary>
    /// <param name="Body">Pet object that needs to be added to the store</param>
    
    /// <returns></returns>
    public void  updatePet (Pet Body) {
      // create path and map variables
      var path = "/pet".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(void) == typeof(byte[])) {
          
          
          _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return;
          
        } else {
          
          
          _apiInvoker.InvokeApi(BasePath, path, "PUT", queryParams, Body, headerParams, formParams);
          return;
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return ;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Add a new pet to the store 
    /// </summary>
    /// <param name="Body">Pet object that needs to be added to the store</param>
    
    /// <returns></returns>
    public void  addPet (Pet Body) {
      // create path and map variables
      var path = "/pet".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(void) == typeof(byte[])) {
          
          
          _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return;
          
        } else {
          
          
          _apiInvoker.InvokeApi(BasePath, path, "POST", queryParams, Body, headerParams, formParams);
          return;
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return ;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Finds Pets by status Multiple status values can be provided with comma seperated strings
    /// </summary>
    /// <param name="Status">Status values that need to be considered for filter</param>
    
    /// <returns></returns>
    public List<Pet>  findPetsByStatus (List<string> Status) {
      // create path and map variables
      var path = "/pet/findByStatus".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      if (Status != null){
        string paramStr = ApiInvoker.ParameterToQueryString(Status);
        queryParams.Add("status", paramStr);
      }
      

      

      

      try {
        if (typeof(List<Pet>) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as List<Pet>;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          if (response != null){
             return (List<Pet>) ApiInvoker.Deserialize(response, typeof(List<Pet>));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Finds Pets by tags Muliple tags can be provided with comma seperated strings. Use tag1, tag2, tag3 for testing.
    /// </summary>
    /// <param name="Tags">Tags to filter by</param>
    
    /// <returns></returns>
    public List<Pet>  findPetsByTags (List<string> Tags) {
      // create path and map variables
      var path = "/pet/findByTags".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      if (Tags != null){
        string paramStr = ApiInvoker.ParameterToQueryString(Tags);
        queryParams.Add("tags", paramStr);
      }
      

      

      

      try {
        if (typeof(List<Pet>) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as List<Pet>;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          if (response != null){
             return (List<Pet>) ApiInvoker.Deserialize(response, typeof(List<Pet>));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Find pet by ID Returns a single pet
    /// </summary>
    /// <param name="PetId">ID of pet to return</param>
    
    /// <returns></returns>
    public Pet  getPetById (long? PetId) {
      // create path and map variables
      var path = "/pet/{petId}".Replace("{format}","json").Replace("{" + "petId" + "}", _apiInvoker.EscapeString(PetId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(Pet) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as Pet;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          if (response != null){
             return (Pet) ApiInvoker.Deserialize(response, typeof(Pet));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Updates a pet in the store with form data 
    /// </summary>
    /// <param name="PetId">ID of pet that needs to be updated</param>
     /// <param name="Name">Updated name of the pet</param>
     /// <param name="Status">Updated status of the pet</param>
    
    /// <returns></returns>
    public void  updatePetWithForm (long? PetId, string Name, string Status) {
      // create path and map variables
      var path = "/pet/{petId}".Replace("{format}","json").Replace("{" + "petId" + "}", _apiInvoker.EscapeString(PetId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      if (Name != null){
        if(Name is byte[]) {
          formParams.Add("name", Name);
        } else {
          string paramStr = (Name is DateTime) ? ((DateTime)(object)Name).ToString("u") : Convert.ToString(Name);
          formParams.Add("name", paramStr);
        }
      }
      if (Status != null){
        if(Status is byte[]) {
          formParams.Add("status", Status);
        } else {
          string paramStr = (Status is DateTime) ? ((DateTime)(object)Status).ToString("u") : Convert.ToString(Status);
          formParams.Add("status", paramStr);
        }
      }
      

      try {
        if (typeof(void) == typeof(byte[])) {
          
          
          _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return;
          
        } else {
          
          
          _apiInvoker.InvokeApi(BasePath, path, "POST", queryParams, null, headerParams, formParams);
          return;
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return ;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Deletes a pet 
    /// </summary>
    /// <param name="ApiKey"></param>
     /// <param name="PetId">Pet id to delete</param>
    
    /// <returns></returns>
    public void  deletePet (string ApiKey, long? PetId) {
      // create path and map variables
      var path = "/pet/{petId}".Replace("{format}","json").Replace("{" + "petId" + "}", _apiInvoker.EscapeString(PetId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      headerParams.Add("api_key", ApiKey);
      

      

      try {
        if (typeof(void) == typeof(byte[])) {
          
          
          _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return;
          
        } else {
          
          
          _apiInvoker.InvokeApi(BasePath, path, "DELETE", queryParams, null, headerParams, formParams);
          return;
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return ;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// uploads an image 
    /// </summary>
    /// <param name="PetId">ID of pet to update</param>
     /// <param name="AdditionalMetadata">Additional data to pass to server</param>
     /// <param name="File">file to upload</param>
    
    /// <returns></returns>
    public ApiResponse  uploadFile (long? PetId, string AdditionalMetadata, byte[] File) {
      // create path and map variables
      var path = "/pet/{petId}/uploadImage".Replace("{format}","json").Replace("{" + "petId" + "}", _apiInvoker.EscapeString(PetId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      if (AdditionalMetadata != null){
        if(AdditionalMetadata is byte[]) {
          formParams.Add("additionalMetadata", AdditionalMetadata);
        } else {
          string paramStr = (AdditionalMetadata is DateTime) ? ((DateTime)(object)AdditionalMetadata).ToString("u") : Convert.ToString(AdditionalMetadata);
          formParams.Add("additionalMetadata", paramStr);
        }
      }
      if (File != null){
        if(File is byte[]) {
          formParams.Add("file", File);
        } else {
          string paramStr = (File is DateTime) ? ((DateTime)(object)File).ToString("u") : Convert.ToString(File);
          formParams.Add("file", paramStr);
        }
      }
      

      try {
        if (typeof(ApiResponse) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as ApiResponse;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "POST", queryParams, null, headerParams, formParams);
          if (response != null){
             return (ApiResponse) ApiInvoker.Deserialize(response, typeof(ApiResponse));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    
  }
  
}
